src/App.tsx

```diff
@@ -1,61 +1,152 @@
 import './App.scss';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const userWithProperty = (property: keyof User, value: string | number) => {
+  return usersFromServer.find(user => user[property] === value) || null;
+};
+
+const userWithTodo = (todo: Todo) => {
+  return userWithProperty('id', todo.userId);
+};
+
+const getTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: userWithTodo(todo),
+}));
+
+const getTodo = (
+  selectedUser: User,
+  title: string,
+  todos: Todo[],
+): Todo => {
+  const lastId = Math.max(...todos.map(todo => todo.id));
+
+  return ({
+    id: lastId + 1,
+    title,
+    completed: false,
+    userId: selectedUser.id,
+    user: selectedUser,
+  });
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getTodos);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
```

Comment:
it's a `boolean` type, so, start name with `has`, `is`
```suggestion
  const [isTitleError, setIsTitleError] = useState(false);
```